metrics-operator/pkg/metrics/server_test.go

```diff
@@ -28,25 +63,34 @@ func TestMetricServer_happyPath(t *testing.T) {
 			Query:                "query",
 			FetchIntervalSeconds: 5,
 		},
+		Status: metricsapi.KeptnMetricStatus{
+			Value:    "12",
+			RawValue: nil,
+			LastUpdated: v1.Time{
+				Time: time.Now(),
+			},
+		},
 	}
-
-	err := metricsapi.AddToScheme(scheme.Scheme)
-	require.Nil(t, err)
 	k8sClient := fake.NewClientBuilder().WithScheme(scheme.Scheme).WithObjects(&metric).Build()
 
-	ctx, cancel := context.WithCancel(context.Background())
-
-	StartServerManager(ctx, k8sClient, openfeature.NewClient("klt-test"), true, 3*time.Second)
+	tInstance := &serverManager{
+		ticker:        clock.New().Ticker(3 * time.Second),
+		ofClient:      openfeature.NewClient("klt-test"),
+		exposeMetrics: true,
+		k8sClient:     k8sClient,
+	}
+	tInstance.start(context.Background())
 
 	require.Eventually(t, func() bool {
-		return instance.server != nil
-	}, 10*time.Second, time.Second)
+		return tInstance.server != nil
+	}, 30*time.Second, time.Second)
 
 	var resp *http.Response
+	var err error
 
 	require.Eventually(t, func() bool {
 		cli := &http.Client{}
-		req, err2 := http.NewRequestWithContext(ctx, http.MethodGet, "http://localhost:9999/metrics", nil)
+		req, err2 := http.NewRequestWithContext(context.TODO(), http.MethodGet, "http://localhost:9999/metrics", nil)
 		require.Nil(t, err2)
```

Comment:
getting an error here should not happen theoretically, but to avoid confusion we should remove the require here